tests/subsys/can/frame/src/main.c

```diff
@@ -77,6 +77,39 @@ static void test_zcan_frame_to_can_frame(void)
 		      "CAN msg length not same");
 }
 
+static void test_invalid_zcan_frame_to_can_frame(void)
+{
+	struct can_frame frame = { 0 };
+	struct can_frame expected = { 0 };
+	struct zcan_frame msg = { 0 };
+	const u8_t data[CAN_MAX_DLEN] = { 0x01, 0x02, 0x03, 0x04,
+					  0x05, 0x06, 0x07, 0x08 };
+
+	expected.can_id = BIT(30) | 0x678;
+	expected.can_dlc = sizeof(data);
+	memcpy(expected.data, data, sizeof(expected.data));
+
+	msg.rtr = 1U;
+	msg.id_type = 0U;
+	msg.ext_id = 0x12345678U;
+	msg.dlc = sizeof(data);
+	memcpy(msg.data, data, sizeof(data));
+
+	can_copy_zframe_to_frame(&msg, &frame);
+
+	LOG_HEXDUMP_DBG((const u8_t *)&frame, sizeof(frame), "frame");
+	LOG_HEXDUMP_DBG((const u8_t *)&msg, sizeof(msg), "msg");
+	LOG_HEXDUMP_DBG((const u8_t *)&expected, sizeof(expected), "expected");
+
+	zassert_mem_equal(&frame.can_id, &expected.can_id, sizeof(frame.can_id),
```

Comment:
You are comparing different types here. zframe ID is a bitfield, and the RTR bit could be anywhere. It's not guaranteed that it's on bit position 30. You should use zassert_equal and mask expected.can_id and test the RTR individually

Reply:
You're totally right.
I've removed it because I think that it won't be any added value to the test.